### ISSUE TITLE
Configuration check triggered every single job in our jenkins environment

### ISSUE THREAD

**Issue by gabriellezbautista** [Nerdwin15/stash-jenkins-postreceive-webhook#140]
We recently upgraded Stash to 3.9.1 and upgraded Stash Jenkins hook to 2.7.0. When we did the check, it ran all the builds

**Comment by ashnazg** [comment]
Every job in your jenkins instance, or every job tied to the one repository whose configuration you were testing?

**Comment by gabriellezbautista** [comment]
Every job in the jenkins instance.

Gabrielle Rafael Z. Bautista
Technology Delivery Center (TDC)
GSCMT Mailbox<mailto:m25622@att.com> | Global SCM Team Wiki<http://wiki.web.att.com/display/GSCMTeam/Global+SCM+Team> | tSpace Community<http://tspace.web.att.com/communities/community/centralscm>

Technology Delivery Transformation (TDT)
Continue transformation of Software Delivery through enhanced processes
and tools that improve quality, productivity and customer satisfaction
________________________________

**Comment by ashnazg** [comment]
I don't see how jobs configured against other repositories could be triggered by a notification from this one repository... I'd think that to be a Jenkins bug, unless its hook API is such that it does indeed accept a call in a format that means "trigger everything, without regard to target repo".

**Comment by mikesir87** [comment]
Agreed.  I'm not sure how the notification could do that.  Do you have any access logs to your Jenkins instance showing the URL being sent?

**Comment by gabriellezbautista** [comment]
I will ask the Jenkins admin (I am the Stash admin) :)

**Comment by diakula** [comment]
We get something similar:
- we have a repo with multiple Jenkins jobs.
- many depend on each other hence do not have any Poll SCM checked.
- many have periodic build set.

Since upgrade we get all jobs triggering on commit.
Jenkins ver. 1.596.1
Atlassian Bitbucket v4.1.3

**Comment by comic96** [comment]
We are also experiencing this - we have many jobs pointing to say, a branch named "develop", only one of which has Poll SCM checked. When someone pushes to the branch, all the jobs are triggered, rather than just the job with Poll SCM enabled.
